libp2p-identify/src/lib.rs

```diff
@@ -49,31 +49,32 @@ mod structs_proto;
 /// Prototype for an upgrade to the identity protocol.
 #[derive(Debug, Clone)]
 pub struct IdentifyProtocol {
-	pub information: IdentifyInfo,
-}
-
-impl IdentifyProtocol {
-	/// Builds a new `IdentifyProtocol`.
-	#[inline]
-	pub fn new(information: IdentifyInfo) -> IdentifyProtocol {
-		IdentifyProtocol {
-			information
-		}
-	}
+	/// Our public key to report to the remote.
+	pub public_key: Vec<u8>,
+	/// Version of the "global" protocol, eg. `ipfs/1.0.0` or `polkadot/1.0.0`.
+	pub protocol_version: String,
+	/// Name and version of the client. Can be thought as similar to the `User-Agent` header
+	/// of HTTP.
+	pub agent_version: String,
+	/// Addresses that we are listening on.
+	pub listen_addrs: Vec<Multiaddr>,
+	/// Protocols supported by us.
+	pub protocols: Vec<String>,
 }
 
 /// Information sent from the listener to the dialer.
 #[derive(Debug, Clone)]
 pub struct IdentifyInfo {
 	/// Public key of the node.
 	pub public_key: Vec<u8>,
-	/// Version of the "global" protocol, eg. `ipfs/1.0.0`.
+	/// Version of the "global" protocol, eg. `ipfs/1.0.0` or `polkadot/1.0.0`.
 	pub protocol_version: String,
-	/// Name and version. Can be thought as similar to the `User-Agent` header of HTTP.
+	/// Name and version of the client. Can be thought as similar to the `User-Agent` header
+	/// of HTTP.
 	pub agent_version: String,
-	/// Addresses that are listened on.
+	/// Addresses that the remote is listening on.
 	pub listen_addrs: Vec<Multiaddr>,
-	/// Address that the server uses to communicate with the dialer.
+	/// Our own address as reported by the remote.
 	pub observed_addr: Multiaddr,
```

Comment:
Bikeshed, but maybe `our_addr`? `observed` makes it sound like we observed it.

Reply:
I went with this name because it matches the protobuf definition.